### PR TITLE
Feat added watchlater,liked video,history,video page

### PR DESCRIPTION
**Pull request by ianuragsharma**
features added :
Single video page,
watch later
like videos
history

### CODE REVIEW

File: src/components/videoCard/VideoCard.jsx [hunk 1]
```diff
@@ -33,17 +45,17 @@ const VideoCard = ({ video }) => {
...
-            </span>
-            <span className="video-uplodaded">{dateUploaded}</span>
-          </div>
-          <p className="video-title fw-500 text-base">{textOverflow(title)}</p>
-          <p className="text-sm  video-alt"> {creator}</p>
+      </Link>
+      <div className="videoText-container ">
+        <div className="text-sm video-alt flex-row">
+          <span>
+            <AiOutlineFire color="#9c9c9c" />
```
Comment (anupkgurung) [inline src/components/videoCard/VideoCard.jsx:54]: why the use of color and not class?

File: src/pages/watchLater/WatchLaterPage.jsx [hunk 2]
```diff
@@ -1,10 +1,20 @@
 import { Navbar } from "../../components";
+import ActionVideoCard from "../../components/actionVideoCard/ActionVideoCard";
```
Comment (anupkgurung) [inline src/pages/watchLater/WatchLaterPage.jsx]: ActionVideoCard can be imported from components where navbar is.

**Review by anupkgurung — APPROVED** [review]
i see you have multiple files for every operation in utils what you can do is to compile it all in one file unless you have to scale any specific file later.
left a couple of suggestion.
rest seems good.

File: src/components/actionVideoCard/ActionVideoCard.jsx [hunk 3]
```diff
@@ -0,0 +1,115 @@
+import "./actionVideoCard.css";
+import React from "react";
+import { AiOutlineLike, AiFillLike } from "react-icons/ai";
+import { BiTimeFive, BiHistory } from "react-icons/bi";
+import { MdOutlineWatchLater, MdWatchLater } from "react-icons/md";
+
+import { RiPlayListLine, RiChatHistoryFill } from "react-icons/ri";
+import { Link, useNavigate } from "react-router-dom";
+import { useAuth, useVideo } from "../../context";
+import { isAlreadyInHistory } from "../../utils/isAlreadyInHistory";
```
Comment (arpit01923) [inline src/components/actionVideoCard/ActionVideoCard.jsx]: use barrel exports here

**Review by arpit01923 — APPROVED** [review]
- clean UI
- code works fine
- one suggestion, try absolute paths

File: src/components/actionVideoCard/ActionVideoCard.jsx (hunk 3 above)
Comment (ianuragsharma) [reply src/components/actionVideoCard/ActionVideoCard.jsx]: will try it.

File: src/pages/watchLater/WatchLaterPage.jsx (hunk 2 above)
Comment (ianuragsharma) [reply src/pages/watchLater/WatchLaterPage.jsx]: thanks for pointing out.

File: src/components/videoCard/VideoCard.jsx (hunk 1 above)
Comment (ianuragsharma) [reply src/components/videoCard/VideoCard.jsx:54]: i have used default color prop in the icon provider so giving it a class here will not work.

**Comment by ianuragsharma** [comment]
> i see you have multiple files for every operation in utils what you can do is to compile it all in one file unless you have to scale any specific file later. left a couple of suggestion. rest seems good.

yes true,the boilerplate for all these code is same.Will work on that

**Pull request merged**